lib/ecto/adapters/firebird/connection.ex

```diff
@@ -152,18 +152,19 @@ defmodule Ecto.Adapters.Firebird.Connection do
   def all(query, as_prefix \\ []) do
     sources = create_names(query, as_prefix)
 
+    limit = limit(query, sources)
+    offset = offset(query, sources)
+
     cte = cte(query, sources)
     from = from(query, sources)
-    select = select(query, sources)
+    select = select(query, limit, offset, sources)
```

Comment:
Rollback this change:

```suggestion
    select = select(query, sources)
```